internal/db/db_blueprints.go

```diff
@@ -24,6 +25,18 @@ const (
 		WHERE blueprints.id = $1 AND blueprints.org_id = $2 AND blueprints.account_number = $3    
 		ORDER BY blueprint_versions.created_at DESC LIMIT 1`
 
+	sqlUpdateBlueprint = `
+		INSERT INTO blueprint_versions (id, blueprint_id, version, body)
+		SELECT 
+			$1,
+			$2,
+			MAX(version) + 1, 
+			$3
+		FROM 
+			blueprint_versions
+		WHERE 
+			blueprint_id = $2`
```

Comment:
I believe we should be checking for org_id her so that when an attacker crafts a valid UUID without being signed into the correct organization, the insert would do nothing:

```
test=> create table test (id serial primary key, text text);
CREATE TABLE
test=> insert into test (text) values ('test');
INSERT 0 1
test=> insert into test (text) select text from test where id = -1;
INSERT 0 0
```

Reply:
Yeah, I believe we should return 404 on access of the user with a different org id / account number. Thanks, I had this in mind, but forgot to implement it. :+1: Definitely legit.

Reply:
Can you add a WHERE clause tho? Just in case someone copy-paste this SQL code, we should definitely check orgs at the database level. Join is required but not big of a deal.

You don’t have to check for this case in handler, I don’t care what an attacker see, it can be just 500. We just care about regular users :-)

Reply:
I think I understand why you would check twice...but I just want to check if you are aware that I run the update in the transaction, where I already check the org_id for blueprint?
And in case it does not succeed, we return error from database.
So if I understand correctly, we want to do a join in case someone use this from outside the transaction?

Reply:
if I understand the resoning, it's to prevent copy paste errors if someone would copy your query and used it standalone outside of the transaction.
I guess we could just add it in the variable name to make it obvious smthing like `sqlOrgUnboundUpdateBlueprintVersion` ? :shrug: